perfkitbenchmarker/linux_benchmarks/hadoop_terasort_benchmark.py

```diff
@@ -12,136 +12,145 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""Runs terasort on hadoop.
+"""Runs a jar using a cluster that supports Apache Hadoop MapReduce.
 
-Cluster Setup:
-http://hadoop.apache.org/docs/stable/hadoop-project-dist/hadoop-common/ClusterSetup.html
+This benchmark takes runs the Apache Hadoop MapReduce Terasort benchmark on an
+Hadoop YARN cluster. The cluster can be one supplied by a cloud provider,
+such as Google's Dataproc or Amazon's EMR.
 
-TODO(user): Make hadoop scale when the number of nodes changes. Also
-investigate other settings and verfiy that we are seeing good performance.
+It records how long each phase (generate, sort, validate) takes to run.
+For each phase, it reports the wall clock time, but this number should
+be used with caution, as it some platforms (such as AWS's EMR) use polling
+to determine when the job is done, so the wall time is inflated
+Furthermore, if the standard output of the job is retrieved, AWS EMR's
+time is again inflated because it takes extra time to get the output.
+
+If available, it will also report a pending time (the time between when the
+job was received by the platform and when it ran), and a runtime, which is
+the time the job took to run, as reported by the underlying cluster.
+
+For more on Apache Hadoop, see: http://hadoop.apache.org/
 """
 
-import logging
-import posixpath
-import time
+import datetime
 
 from perfkitbenchmarker import configs
-from perfkitbenchmarker import flags
 from perfkitbenchmarker import sample
-from perfkitbenchmarker import vm_util
-from perfkitbenchmarker.linux_packages import hadoop
+from perfkitbenchmarker import spark_service
+from perfkitbenchmarker import flags
 
-flags.DEFINE_integer('terasort_num_rows', 100000000,
-                     'Number of 100-byte rows used in terasort.')
 
-FLAGS = flags.FLAGS
 
 BENCHMARK_NAME = 'hadoop_terasort'
 BENCHMARK_CONFIG = """
 hadoop_terasort:
-  description: Runs Terasort. Control the number of worker VMs with --num_vms.
-  vm_groups:
-    master:
-      vm_spec: *default_single_core
-      disk_spec: *default_500_gb
-    workers:
-      vm_spec: *default_single_core
-      disk_spec: *default_500_gb
-      vm_count: 8
+  description: Run the Apache Hadoop MapReduce Terasort benchmark on a cluster.
+  spark_service:
+    service_type: pkb_managed
+    worker_group:
+      vm_spec:
+        GCP:
+          machine_type: n1-standard-4
+          boot_disk_size: 500
```

Comment:
may make sense to add either the vm specs or entire spark spec to the default constants yaml file so you can just use the * notation like in "*default_single_core"